Add log sharing message on login.

dchw opened the pull request:
~~Hold until after our next release.~~

alexcb commented on cmd/earthly/main.go:
```diff
@@ -2492,21 +2495,28 @@ func (app *earthlyApp) actionAccountLogin(c *cli.Context) error {
 }
 
+func (app *earthlyApp) printLogSharingMessage() {
+	app.console.Printf("Log sharing is enabled by default. If you would like to disable it, run:")
```
nit: move `\n` from below printf up to this line.

dchw replied:
I smooshed all the printing together with the intent on making the resulting layout/formatting more explicit.

dchw commented:
@alexcb @brandonSc I flipped a few more switches that made sense with this PR.

1. Log sharing is default now
2. Removed un-disabling sharing in GHA actions
3. Fixed non-logged in users message (bad closure + `errors.Is`)

brandonSc reviewed (approved):
🎉
